pkg/dyninst/irgen/irgen.go

```diff
@@ -140,24 +140,38 @@ func generateIR(
 		cfg.maxDynamicTypeSize,
 		cfg.maxHashBucketsSize,
 	)
-	dwarfRes, err := processDwarf(interests, d, typeCatalog, objFile)
+	pendingSubprograms, err := processDwarf(interests, d, typeCatalog, objFile)
 	if err != nil {
 		return nil, err
 	}
-	pendingSubprograms := dwarfRes.pendingSubprograms
-	abstractIdx := dwarfRes.firstAbstractIdx
 
-	// Determine injection points right after each subprogram's prologue.
-	prologueResults, err := computePrologueResults(
-		objFile, pendingSubprograms, abstractIdx,
-	)
+	// Find prologues need to determine injection points. We make an assumption
+	// that prologue, if function has a frame, should be contained within the
+	// first pc range of a subprogram. This simplifies the logic slightly.
+	prologueSearch := make([]prologueSeachParams, 0, len(pendingSubprograms))
+	for _, sp := range pendingSubprograms {
+		if sp.subprogram.OutOfLinePCRanges != nil {
```

Comment:
It's more correct to check if `len(sp.subprogram.OutOfLinePCRanges) > 0` in case later we add some sort of subslicing or pooling.